src/main/java/de/tum/in/www1/artemis/web/rest/StaticCodeAnalysisResource.java

```diff
@@ -0,0 +1,161 @@
+
+    private static final String ENTITY_NAME = "StaticCodeAnalysisCategory";
+
+    private final Logger log = LoggerFactory.getLogger(StaticCodeAnalysisResource.class);
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final StaticCodeAnalysisService staticCodeAnalysisService;
+
+    public StaticCodeAnalysisResource(AuthorizationCheckService authCheckService, ProgrammingExerciseService programmingExerciseService,
+            StaticCodeAnalysisService staticCodeAnalysisService) {
+        this.authCheckService = authCheckService;
+        this.programmingExerciseService = programmingExerciseService;
+        this.staticCodeAnalysisService = staticCodeAnalysisService;
+    }
+
+    /**
+     * Get the static code analysis categories for a given exercise id.
+     *
+     * @param exerciseId of the the exercise
+     * @return the static code analysis categories
+     */
+    @GetMapping(Endpoints.CATEGORIES)
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<StaticCodeAnalysisCategory>> getStaticCodeAnalysisCategories(@PathVariable Long exerciseId) {
+        log.debug("REST request to get static code analysis categories for programming exercise {}", exerciseId);
+
+        ProgrammingExercise programmingExercise = programmingExerciseService.findById(exerciseId);
+
+        if (!Boolean.TRUE.equals(programmingExercise.isStaticCodeAnalysisEnabled())) {
+            return badRequest();
+        }
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(programmingExercise)) {
+            return forbidden();
+        }
+
+        Set<StaticCodeAnalysisCategory> staticCodeAnalysisCategories = staticCodeAnalysisService.findByExerciseId(exerciseId);
+        return ResponseEntity.ok(staticCodeAnalysisCategories);
+    }
+
+    /**
+     * Updates the static code analysis categories of a given programming exercise using the data in the request body.
+     *
+     * @param exerciseId of the the exercise
+     * @param categories used for the update
+     * @return the updated static code analysis categories
+     */
+    @PatchMapping(Endpoints.CATEGORIES)
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<StaticCodeAnalysisCategory>> updateStaticCodeAnalysisCategories(@PathVariable Long exerciseId,
+            @RequestBody Set<StaticCodeAnalysisCategory> categories) {
+        log.debug("REST request to update static code analysis categories for programming exercise {}", exerciseId);
+
+        ProgrammingExercise programmingExercise = programmingExerciseService.findById(exerciseId);
+
+        if (!Boolean.TRUE.equals(programmingExercise.isStaticCodeAnalysisEnabled())) {
+            return badRequest();
+        }
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(programmingExercise)) {
```

Comment:
Should TAs be allowed to update the categories or is this a task that only instructors can perform?

Reply:
At the moment test cases can be changed also by TA. As the categories will be changed in the same view, I've chosen to use the same settings.